### PR TITLE
[WIP] coreutils: 8.31 -> 8.32

### PR DESCRIPTION
**Pull request by lovesegfault**
###### Motivation for this change
While working on #85951 I noticed this was out-of-date.

cc. @FRidh for inclusion in staging.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/coreutils/default.nix [hunk 1]
```diff
@@ -17,21 +17,14 @@ with lib;
...
 
-  patches = optional stdenv.hostPlatform.isCygwin ./coreutils-8.23-4.cygwin.patch
-         # Fix failing test with musl. See https://lists.gnu.org/r/coreutils/2019-05/msg00031.html
-         # To be removed in coreutils-8.32.
-         ++ optional stdenv.hostPlatform.isMusl ./avoid-false-positive-in-date-debug-test.patch
-         # Fix compilation in musl-cross environments. To be removed in coreutils-8.32.
-         ++ optional stdenv.hostPlatform.isMusl ./coreutils-8.31-musl-cross.patch
-         # Fix compilation in android-cross environments. To be removed in coreutils-8.32.
-         ++ [ ./coreutils-8.31-android-cross.patch ];
+  patches = optional stdenv.hostPlatform.isCygwin ./coreutils-8.23-4.cygwin.patch;
```
Comment (veprbl) [inline pkgs/tools/misc/coreutils/default.nix:27]: Did you check that these don't apply anymore or just removed based on the comment?

Comment (lovesegfault) [reply pkgs/tools/misc/coreutils/default.nix:27]: The latter

Comment (veprbl) [reply pkgs/tools/misc/coreutils/default.nix:27]: The remaining patch doesn't apply
cc @tobim

Comment (veprbl) [reply pkgs/tools/misc/coreutils/default.nix:27]: @tobim , sorry for the noise.
The patch came in at 221b5a5be5eb698c696b728b0a9b6646422cd9c7

It doesn't apply against 8.31, so I propose to remove cygwin-specific things.

Comment (lovesegfault) [reply pkgs/tools/misc/coreutils/default.nix]: Done! Nice catch @veprbl.

Comment (veprbl) [reply pkgs/tools/misc/coreutils/default.nix]: @lovesegfault Could you, please, remove the other bit too?
https://github.com/NixOS/nixpkgs/blob/51fa4f0d9a9d6491d9b39a0d1e34f804bd05587c/pkgs/tools/misc/coreutils/default.nix#L67
If the patch is gone, nothing will change configure.ac, so there is no need to regenerate.

Comment (lovesegfault) [reply pkgs/tools/misc/coreutils/default.nix]: Done, removed the `musl` part too as that patch is now gone as well.

**Comment by FRidh** [comment]
aarch64 failure

**Comment by lovesegfault** [comment]
> aarch64 failure

Fix incoming

**Comment by lovesegfault** [comment]
Alright fixed the issues on AArch64, also re-enabled parallel building as the noted issue for having it disabled is no longer reproducible.

cc. @FRidh

**Comment by veprbl** [comment]
@GrahamcOfBorg build coreutils pkgsMusl.coreutils

**Comment by veprbl** [comment]
@GrahamcOfBorg build coreutils
@GrahamcOfBorg build pkgsMusl.coreutils
@GrahamcOfBorg build pkgsi686Linux.coreutils

File: pkgs/tools/misc/coreutils/getdents64.patch [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+             break;
+         }
+-#ifdef __linux__
+       else if (! found_any_entries)
+         {
++#if defined __linux__ && defined SYS_getdents64
+           /* If readdir finds no directory entries at all, not even "." or
+              "..", then double check that the directory exists.  */
+-          if (syscall (SYS_getdents, dirfd (dirp), NULL, 0) == -1
++          if (syscall (SYS_getdents64, dirfd (dirp), NULL, 0) == -1
```
Comment (veprbl) [inline pkgs/tools/misc/coreutils/getdents64.patch:28]: Perhaps we should rather fetch https://github.com/coreutils/coreutils/commit/10fcb97bd728f09d4a027eddf8ad2900f0819b0a

File: pkgs/tools/misc/coreutils/default.nix [hunk 3]
```diff
@@ -17,21 +17,14 @@ with lib;
```
Comment (veprbl) [inline pkgs/tools/misc/coreutils/default.nix:17]: Drop `bison` from argument list

**Comment by veprbl** [comment]
cc @bkchr as author of 320a4c15265d6afb7eb6e28fcc1f72a80367ccc1

**Comment by lovesegfault** [comment]
I'm working on this some more

**Pull request closed without merging**